src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -11,108 +11,84 @@
+namespace Microsoft.CodeAnalysis.CSharp.NewLines.ConditionalExpressionPlacement;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    internal sealed class ConditionalExpressionPlacementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    public ConditionalExpressionPlacementDiagnosticAnalyzer()
+        : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
+               EnforceOnBuildValues.ConditionalExpressionPlacement,
+               CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
+               new LocalizableResourceString(
+                   nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
     {
-        public ConditionalExpressionPlacementDiagnosticAnalyzer()
-            : base(IDEDiagnosticIds.ConditionalExpressionPlacementDiagnosticId,
-                   EnforceOnBuildValues.ConditionalExpressionPlacement,
-                   CSharpCodeStyleOptions.AllowBlankLineAfterTokenInConditionalExpression,
-                   new LocalizableResourceString(
-                       nameof(CSharpAnalyzersResources.Blank_line_not_allowed_after_conditional_expression_token), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
-        {
-        }
+    }
 
-        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
-            => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
+    public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+        => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
 
-        protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterCompilationStartAction(context =>
-                context.RegisterSyntaxTreeAction(treeContext => AnalyzeTree(treeContext, context.Compilation.Options)));
+    protected override void InitializeWorker(AnalysisContext context)
+        => context.RegisterSyntaxNodeAction(ProcessConditionalExpression, SyntaxKind.ConditionalExpression);
 
-        private void AnalyzeTree(SyntaxTreeAnalysisContext context, CompilationOptions compilationOptions)
-        {
-            var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInConditionalExpression;
-            if (option.Value || ShouldSkipAnalysis(context, compilationOptions, option.Notification))
-                return;
+    private void ProcessConditionalExpression(SyntaxNodeAnalysisContext context)
+    {
+        var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInConditionalExpression;
+        if (option.Value || ShouldSkipAnalysis(context, option.Notification))
```

Comment:
@mavasani it would be really good if we could register a start action for syntax node actions that would allow doing these skip-analysis tests when hitting the file the first time.

Reply:
You can use a symbol start action, and register the syntax node action within that.

Reply:
i considered this, but symbol actions aren't fun.  Most specifically because they then have to deal with teh complexities of partial types and what that means.  My preference is to keep this somehow *solely* file-specific and have hte semantics of checking that file against teh user's config settings be 100% clear.